data-avro/src/main/java/com/linkedin/data/avro/SchemaTranslator.java

```diff
@@ -341,6 +354,26 @@ public static String dataToAvroSchemaJson(DataSchema dataSchema, DataToAvroSchem
     return SchemaToAvroJsonEncoder.schemaToAvro(translatedDataSchema, dataSchema, defaultValueOverrides, options);
   }
 
+  /**
+   * Adds TRANSLATED_FROM_SOURCE_OPTION property to named data schemas if not already present.
+   * @param dataSchema the data schema to add the property to
+   * @return the data schema with the property added if applicable.
+   */
+  private static DataSchema addTranslatedPropToNamedDataSchema(DataSchema dataSchema) {
+    // Add translated from annotation if this is a named dataSchema
+    if (dataSchema instanceof NamedDataSchema) {
+      NamedDataSchema namedDataSchema = (NamedDataSchema) dataSchema;
+      // Add annotation if not already present
+      if (!namedDataSchema.getProperties().containsKey(TRANSLATED_FROM_SOURCE_OPTION)) {
```

Comment:
what should you do if existing translated_from prop is different from updated one?

Reply:
It doesn't update the source in that case. So, if schema `A` translates to `B`, and `B` is translated to `C`,
the translatedFrom option in C will be `A`

Reply:
My original question is talking about an error case, I am guessing that should never happen and we should throw error? 
But your response provides me something  different from my understanding, I thought that annotation is just for direct source, seems not?

Reply:
We don't need to stop users from translating a translated schema again, as it is allowed right now in SchemaTranslator right now, and stopping it might fail users.

However, we can use this annotation to fail other flows like PDL -> Proto or Avro -> proto.

Reply:
Based on this logic, `translated.from` annotation is used to indicate the root source, not directly translated source, is this clearly communicated?